python/taichi/lang/kernel_impl.py

```diff
@@ -841,26 +842,33 @@ def call_back():
                 runtime_ops.sync()
 
             if has_ret:
-                if id(ret_dt) in primitive_types.integer_type_ids:
-                    if is_signed(cook_dtype(ret_dt)):
-                        ret = t_kernel.get_ret_int(0)
-                    else:
-                        ret = t_kernel.get_ret_uint(0)
-                elif id(ret_dt) in primitive_types.real_type_ids:
-                    ret = t_kernel.get_ret_float(0)
-                elif id(ret_dt.dtype) in primitive_types.integer_type_ids:
-                    if is_signed(cook_dtype(ret_dt.dtype)):
-                        it = iter(t_kernel.get_ret_int_tensor(0))
-                    else:
-                        it = iter(t_kernel.get_ret_uint_tensor(0))
-                    ret = Matrix([[next(it) for _ in range(ret_dt.m)]
-                                  for _ in range(ret_dt.n)],
-                                 ndim=getattr(ret_dt, 'ndim', 2))
+                taichi_arch = impl.current_cfg().arch
+                if taichi_arch == _ti_core.Arch.x64 or taichi_arch == _ti_core.Arch.arm64 \
+                        or taichi_arch == _ti_core.Arch.cuda or taichi_arch == _ti_core.Arch.amdgpu:
```

Comment:
do we have a `arch_is_llvm` thing in python scope?

Reply:
Let me add one.